### PR TITLE
Use new oci compat mediatype for bundles

### PR DESCRIPTION
**Pull request by loosebazooka**

### CODE REVIEW

File: sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java [hunk 1]
```diff
@@ -59,8 +59,10 @@ class BundleFactoryInternal {
   private static final String BUNDLE_V_0_1 = "application/vnd.dev.sigstore.bundle+json;version=0.1";
   private static final String BUNDLE_V_0_2 = "application/vnd.dev.sigstore.bundle+json;version=0.2";
   private static final String BUNDLE_V_0_3 = "application/vnd.dev.sigstore.bundle+json;version=0.3";
+  // media_type format switch: https://github.com/sigstore/protobuf-specs/pull/279
+  private static final String BUNDLE_V_0_3_1 = "application/vnd.dev.sigstore.bundle.v0.3+json";
```
Comment (haydentherapper) [inline sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java:63]: To confirm, is there anywhere in the verification path where you need to treat `BUNDLE_V_0_3_1` and `BUNDLE_V_0_3` as equivalent bundles?

**Review by haydentherapper — APPROVED** [review]

File: sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java (hunk 1 above)
Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java:63]: equivalent? We just parse them or we dont.

Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java:63]: The reality of this is we only have a single parser based on the protobuf spec build at v0.3.1, we're pretty permissive with the bundle mediatype. We don't punish v2, v1 bundles that use v3 features. If the v3 parser can parse it, we're okay and we use it.

Comment (loosebazooka) [reply sigstore-java/src/main/java/dev/sigstore/bundle/BundleFactoryInternal.java:63]: In fact, we ignore that v1 allows no inclusion proof. Those bundles are invalid as far as we're concerned.

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**